retry/api.py

```diff
@@ -40,8 +40,9 @@ def __retry_internal(f, exceptions=Exception, tries=-1, delay=0, max_delay=None,
                     func_qualname =  f.func.__qualname__
                 except AttributeError:
                     func_qualname = str(f.func)
-                logger.warning('%s: %s in %s.%s, retrying in %s seconds...', e.__class__.__qualname__, e,
-                               f.func.__module__, func_qualname, _delay)
+                logger.warning('{}: {} in {}.{}, retrying in {} seconds...'.format(
```

Comment:
Using early-formatting is considered as a bad practice for Python logging. This would generate even linter warnings.
Please check [this small blog](https://christopher.xyz/2019/02/07/python-logging-interpolation.html) and [this paragraph of logging cookbook](https://docs.python.org/3/howto/logging-cookbook.html#use-of-alternative-formatting-styles)  for more detailed information on this.